### ISSUE TITLE
[BUG] max has a bad error message

### ISSUE THREAD

**Issue by DenisYaroshevskiy** [jfalcou/eve#1714]
If you think that `max(x)`  does reduce(x, max) - you get a confusing error message. needs a concept.

https://godbolt.org/z/vv7oWMzs1

**Comment by jfalcou** [comment]
It now returns an arror saying that max(x) returns a lambda while we want a vlaue.

The issue is that we can't have a concept for random callable with unspecified type.
Maybe using `!eve::value` is enough ?

**Comment by jfalcou** [comment]
On fix/apply-over, the error is :

```c++
<source>:8:12: error: no matching function for call to object of type 'const max_t<eve::options<>>'
    return eve::max(x-8);
           ^~~~~~~~
/home/falcou/dev/eve/include/eve/module/core/regular/max.hpp:30:5: note: candidate template ignored: substitution failure [with Tup = wide<float, fixed<8>>]: constraints not satisfied for alias template 'apply_traits_t' [with Traits = eve::common_value, Tuple = eve::wide<float, eve::fixed<8>>]
    operator()(Tup const & t) const noexcept  requires(kumi::size_v<Tup> >= 2) { return EVE_DISPATCH_CALL(t); }
    ^
/home/falcou/dev/eve/include/eve/module/core/regular/max.hpp:34:36: note: candidate template ignored: constraints not satisfied [with Callable = wide<float, fixed<8>>]
    EVE_FORCEINLINE constexpr auto operator()(Callable const & f) const noexcept { return EVE_DISPATCH_CALL(f); }
                                   ^
/home/falcou/dev/eve/include/eve/module/core/regular/max.hpp:33:47: note: because '!eve::ordered_value<eve::wide<float, eve::fixed<8> > >' evaluated to false
    requires(!kumi::product_type<Callable> && !eve::ordered_value<Callable>)
                                              ^
/home/falcou/dev/eve/include/eve/module/core/regular/max.hpp:20:52: note: candidate function template not viable: requires 2 arguments, but 1 was provided
    EVE_FORCEINLINE constexpr common_value_t<T, U> operator()(T t, U u) const noexcept { return EVE_DISPATCH_CALL(t, u); }
                                                   ^
/home/falcou/dev/eve/include/eve/module/core/regular/max.hpp:23:61: note: candidate function template not viable: requires at least 2 arguments, but 1 was provided
    EVE_FORCEINLINE constexpr common_value_t<T0, T1, Ts...> operator()(T0 t0, T1 t1, Ts...ts) const noexcept
                                                            ^
1 error generated.
```

**Comment by DenisYaroshevskiy** [comment]
I think yeah, lgtm

**Comment by jfalcou** [comment]
I'll close this once fix/apply-over is merged

**Comment by jfalcou** [comment]
Fixed in 20b383f
